### ISSUE TITLE
Can't post on Factorio Forums yet...

### ISSUE THREAD

**Issue by xrobau** [Rainson12/FactorioSaveGameEnableAchievements#1]
This doesn't appear to work when purely enabling the editor. There is no 'command-ran' when simply starting a new game and then /editor.

There IS 'editor-will-disable-achievements', which has '00000001000100000000000a0a00000000ffffffffffff' a fair way back from it, but that's not the 6/7 byte offset you have in your script. Where did you get those numbers from, out of curiosity?

**Comment by xrobau** [comment]
I just threw this together in PHP, as I'm too lazy to bother compiling things:

```
<?php
$f = 'BlankHacked.zip';
$za = new ZipArchive();
$za->open($f);
$ptr = $za->numFiles;
while ($ptr) {
        $name = $za->getNameIndex($ptr);
        $b = basename($name);
        if (preg_match('/level.dat\d+/', $b)) {
                $f = $za->getFromIndex($ptr);
                $r = zlib_decode($f);
                $p = strpos($r, 'editor-will-disable-achievements');
                if ($p !== false) {
                        print "Found it at $p in $b\n";
                        $chunk = substr($r, $p - 4000, 4100);
                        file_put_contents("/tmp/chunk", $chunk);
                        exit;
                }
                $ptr--;
        }
}
```

**Comment by xrobau** [comment]
After messing around with it, it appears to be as simple as this for /editor

```
$h = bin2hex($fixed);
$fixedhex = str_replace(
    [ '00000000000100fffffffffffffffff' ].
    [ '00000000000000fffffffffffffffff' ],
    $h);
```

**Comment by Rainson12** [comment]
Good catch, I will be updating my code. Initially I wrote my Patch to only fix command execution however in my first try I also used the editor without really editing anything hence I found the two 01 01 occurences, without really thinking about how to identify whether the editor was used or if it was only the commands. The patch here: https://github.com/Rainson12/FactorioSaveGameEnableAchievements/blob/master/FactorioSaveGameEnableAchievements/Program.cs#L36 should still be valid only the detection is off.
